### PR TITLE
remove amp factor calls

### PR DESCRIPTION
**Pull request by mendesfabio**
# Description

Removes calls to get the current Amp Factor and calculates it using parameters from the latest update.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [x] I have commented my code where relevant, particularly in hard-to-understand areas

### CODE REVIEW

File: src/mappings/helpers/stable.ts [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
 
 export const AMP_PRECISION = BigInt.fromI32(1000);
 
-export function updateAmpFactor(pool: Pool): void {
-  let poolContract = StablePool.bind(changetype<Address>(pool.address));
+export function updateAmpFactor(pool: Pool, blockTimestamp: BigInt): void {
+  let latestAmpUpdateId = pool.latestAmpUpdate;
+  if (latestAmpUpdateId === null) return;
 
-  pool.amp = getAmp(poolContract);
```
Comment (markusbkoch) [inline src/mappings/helpers/stable.ts:11]: Problem with removing this is that for pools that have ongoing amp updates in progress, which started before the graft base block, their amps won't be updated. I think we need to keep the call as a fallback until we do a full sync.

File: src/mappings/helpers/stable.ts [hunk 2]
```diff
@@ -1,28 +1,43 @@
...
+  let startValue = latestAmpUpdate.startAmp;
+  let endValue = latestAmpUpdate.endAmp;
+  let startTime = latestAmpUpdate.startTimestamp;
+  let endTime = latestAmpUpdate.endTimestamp;
+
+  let value = ZERO;
+  if (blockTimestamp.lt(endTime)) {
+    if (endValue.gt(startValue)) {
+      value = startValue.plus(
+        endValue.minus(startValue).times(blockTimestamp.minus(startTime)).div(endTime.minus(startTime))
```
Comment (markusbkoch) [inline src/mappings/helpers/stable.ts:29]: for increased readability I'd suggest creating consts for `blockTimestamp.minus(startTime)` and `endTime.minus(startTime)`

File: src/mappings/helpers/stable.ts (hunk 1 above)
Comment (markusbkoch) [reply src/mappings/helpers/stable.ts:11]: Or alternatively make sure no pools have amp updates in progress; or if there are a few temporarily hardcode their curves here

File: src/mappings/helpers/stable.ts (hunk 2 above)
Comment (mendesfabio) [reply src/mappings/helpers/stable.ts]: https://github.com/balancer/balancer-subgraph-v2/pull/495/commits/0c209d2432565a7b02d2a770aedbf2093eac9c7c

File: src/mappings/helpers/stable.ts (hunk 1 above)
Comment (mendesfabio) [reply src/mappings/helpers/stable.ts:11]: https://github.com/balancer/balancer-subgraph-v2/pull/495/commits/ad015a164d85db8faac090a7462479a84159b2a1

**Review by markusbkoch — APPROVED** [review]

**Pull request merged**